### DISCUSSION TITLE
Retrieve user id in trigger function

### DISCUSSION THREAD

**Discussion by zarSou9** [supabase/supabase#22769] [Questions]
Hello Supa Community,

I am trying to authorize users to update a row depending on which columns they are updating and their user_id. I **can** easily access the user's id in an rls policy with auth.uid() but I **can't** compare old and new rows (for column level restrictions). However with trigger functions, you **can** compare old and new rows but (from what i've found and tried), **can't** access the id of the user who made the update request. I have tried auth.uid() and by using current setting (current_setting('jwt.claims.sub', true)), but they both don't work. Any suggestions on how I could do this?

**Comment by GaryAustin1** [comment]
auth.uid() will be available on all requests and trigger functions that started with a database API request.   Old/New only contain the information in the columns of your table. 
If you are not getting  auth.uid() with a value in a trigger function then it likely did not originate with an API request to the database (supabase-js).

**Reply by sp88011** [reply]
> If you are not getting auth.uid() with a value in a trigger function then it likely did not originate with an API request to the database (supabase-js).

I'm using Drizzle, and facing the same problem. Would I have to use `supabase-js` in order for `auth.uid()` to be available in a trigger?

My flow
1. Drizzle query to insert a new row in a table
2. Trigger fires a function
3. That function wants to access `auth.uid()` but it is always NULL

**Reply by GaryAustin1** [reply]
Yes you have to use the REST API for auth.uid() to work.  It gets its info from the JWT sent in the authorization header.

You can fake the data auth.uid() uses with SQL.  
This is an example of what I used for RLS testing in SQL:
```sql
set session role authenticated;
set request.jwt.claims to '{"role":"authenticated", "sub":"5950b438-b07c-4012-8190-6ce79e4bd8e5"}';
```

This also has some discussions on RLS with Supabase and Drizzle:  https://orm.drizzle.team/docs/rls#using-with-supabase
